src/common/io/session.c

```diff
@@ -92,9 +106,25 @@ ioSessionIoWrite(IoSession *this)
     FUNCTION_TEST_RETURN(this->interface->ioWrite(this->driver));
 }
 
+/**********************************************************************************************************************************/
+IoSessionRole
+ioSessionRole(const IoSession *this)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(IO_SESSION, this);
+    FUNCTION_TEST_END();
+
+    ASSERT(this != NULL);
+
+    FUNCTION_TEST_RETURN(this->interface->role(this->driver));
+}
+
 /**********************************************************************************************************************************/
 String *
 ioSessionToLog(const IoSession *this)
 {
-    return strNewFmt("{type: %s, driver: %s}", strZ(*this->interface->type), strZ(this->interface->toLog(this->driver)));
+    return strNewFmt(
+        "{type: %s, role: %s, driver: %s}", strZ(*this->interface->type),
+        this->interface->role(this->driver) == ioSessionRoleClient ? "client" : "server",
+        strZ(this->interface->toLog(this->driver)));
```

Comment:
This will be shown in the driver if it is valid, but sure. More info is generally better.

Reply:
Actually, I think it's better just to leave this logged in the `driver: {...}` part. ioSessionFd() requires a non-const pointer so we need to either duplicate the logic or `UNCONSTIFY()` it. Neither is very attractive since the info is already there.